src/cluster/backend/zeromq/ZeroMQ.cc

```diff
@@ -227,7 +218,9 @@ bool ZeroMQBackend::DoInit() {
     // Setup connectivity between main and child thread.
     main_inproc.bind("inproc://inproc-bridge");
     child_inproc.connect("inproc://inproc-bridge");
-    self_thread = std::thread(self_thread_fun, this);
+
+    // Thread is joined in backend->DoTerminate(), backend outlives it.
```

Comment:
Since this is not a `jthread`, but just a `thread` it will not autojoin on destruction; let's add a destructor for `ZeroMQ` so we do not leak this thread under any circumstances. If `DoTerminate` is idempotent we might be able to just call it in the dtr for cleanup (in a `try`/`catch` to prevent throwing from a dtr).

And if I understand correctly, as written this change here just fixes style, but doesn't change behavior at all.

Reply:
Correct, no change in functionality. Let me see about the destruction, thanks!